### DISCUSSION TITLE
Mimesis schema to db

### DISCUSSION THREAD

**Discussion by euri10** [lk-geimfari/mimesis#1197] [Q&A]
greetings, I just discovered this library and so far like it a lot !

I'm using it mainly to feed my test db with fake data and I wondered if the approach I'm taking looks sensible or if there were more interesting ways to do that.

Currently I'm
1. create a schema per table, something like that.
```
            schema = Schema(schema=lambda: {
                "products_id": _("increment"),
                "name": _("text.word"),
                "unit_price": _("price"),
                "image_path": _("hash"),
            })
```
2. save it as a csv `schema.to_csv(file_path='data.csv', iterations=1000)`
3.  copy it to the table `await conn.copy_to_table("products", source="data.csv", format="csv", header=True)`

I was wondering if there was a more automated way to do this, this is currently very fast, the only small issue I have is that it is repetitive with table increases and hard to maintain with schema changes.

**Comment by lk-geimfari** [comment]
If you're using ORM then a better approach is using `Generic` or https://github.com/lk-geimfari/mimesis-factory.

Here is an example of using `Mimesis` with `SQLAlchemy`:

```python
class Patient(db.Model):
    id = db.Column(db.Integer, primary_key=True)
    email = db.Column(db.String(120), unique=True)
    phone_number = db.Column(db.String(25))
    full_name = db.Column(db.String(100))
    weight = db.Column(db.String(64))
    height = db.Column(db.String(64))
    blood_type = db.Column(db.String(64))

    def __init__(self, **kwargs):
        super(Patient, self).__init__(**kwargs)

    @staticmethod
    def _bootstrap(count=2000, locale='en'):
        from mimesis import Generic
        from mimesis.locale import Locale


        generic = Generic(Locale.FR)

        for _ in range(count):
            patient = Patient(
                email=generic.person.email(),
                phone_number=generic.person.telephone(),
                full_name=generic.person.full_name(gender='female'),
                weight=generic.person.weight(),
                height=generic.person.height(),
                blood_type=generic.person.blood_type()
            )

            db.session.add(patient)
            try:
                db.session.commit()
            except Exception:
                db.session.rollback()
```

It's better to store code from `_bootstrap` as a separate command.

I hope, I explained the idea well. Anyway, if you don't get it then don't be shy to ask me — I'll help you.

**Reply by euri10** [reply]
That's a very nice, detailed and interesting explanation thanks !

Unfortunately I've stopped using any orm a while ago so this does not really fit my requirements :) but the approach looks cool.

I feel like the advantage of using `copy_to_table` is that it is fast enough for large dumps of fake data, which is something I really need.

I was probably asking for something "impossible" in the sense not using an orm and wanting to get for free some kind of "mapping" :face_with_spiral_eyes:
